eth/eventhandler/handlers.go

```diff
@@ -120,8 +120,14 @@ func (eh *EventHandler) handleOperatorRemoved(txn basedb.Txn, event *contract.Co
 		fields.Owner(od.OwnerAddress),
 	)
 
-	// This function is currently no-op and it will do nothing. Operator removed event is not used in the current implementation.
+	// Delete the operator data from storage to mark it as removed
+	// This will cause message validation to reject messages from this operator
+	if err := eh.nodeStorage.DeleteOperatorData(txn, event.OperatorId); err != nil {
+		logger.Error("failed to delete operator data", zap.Error(err))
+		return fmt.Errorf("could not delete operator data: %w", err)
```

Comment:
I don't really like that we log AND return the err because it will probably be logged above as well. The only issue that if we don't log here the above log won't contain the added fields (pk, owner, etc..). So I guess we can leave this.

Reply:
good point, fixed!